core/src/main/java/org/apache/accumulo/core/data/ArrayByteSequence.java

```diff
@@ -180,4 +180,19 @@ public byte[] toArray() {
   public String toString() {
     return new String(data, offset, length, UTF_8);
   }
+
+  /**
+   * Resets data, offset, and length of a backing array for this byte sequence object.
+   *
+   * @since 3.1.0
+   */
+  public void reset(byte[] data, int offset, int length) {
```

Comment:
#4735 has been merged into `main`. The reset method now exists, but is missing the bounds checking that you have added here.